crates/hyperdrive-math/src/long/max.rs

```diff
@@ -118,7 +124,15 @@ impl State {
             if maybe_derivative.is_none() {
                 break;
             }
-            let possible_max_base_amount = max_base_amount + solvency / maybe_derivative.unwrap();
+            let mut possible_max_base_amount =
+                max_base_amount + solvency / maybe_derivative.unwrap();
+
+            // possible_max_base_amount might be less than minimum transaction amount.
+            // we clamp here if so
+            if possible_max_base_amount < self.minimum_transaction_amount() {
```

Comment:
Same here

Reply:
Again, I'll keep this check here due to a possibility of the `possible_max_base_amount` overshooting. This clamp is mostly for avoiding a `minimum_transaction_error` in `calculate_open_long`. I'll add a final check and return 0 if this value is too small at the end.